### PR TITLE
[django] Use a set instead of list for cache_backends to avoid duplicates

### PR DESCRIPTION
**Pull request by wenbochang**
Sorry I'm missing something simple, but the list comprehension for cache backends maintains duplicate classes, for example if we have something like

```
    'cache1': {
        'BACKEND': 'django_redis.cache.RedisCache',
        'LOCATION': ...
        'OPTIONS': {
            ...
        }
    },
    'cache2': {
        'BACKEND': 'django_redis.cache.RedisCache',
        'LOCATION': ...
        'OPTIONS': {
            ...
        }
    },
```

This ends logging a lot of "already traced" and can be fixed with a set comprehension.

### CODE REVIEW

File: ddtrace/contrib/django/cache.py [hunk 1]
```diff
@@ -41,7 +41,7 @@ def patch_cache(tracer):
           Django supported cache servers (Redis, Memcached, Database, Custom)
     """
     # discover used cache backends
-    cache_backends = [cache['BACKEND'] for cache in django_settings.CACHES.values()]
+    cache_backends = {cache['BACKEND'] for cache in django_settings.CACHES.values()}
```
[the remark this answers is not in the document]
Comment (wenbochang) [reply ddtrace/contrib/django/cache.py]: sounds good!

**Comment by brettlangdon** [comment]
@wenbochang thanks!

File: ddtrace/contrib/django/cache.py [hunk 2]
```diff
@@ -102,7 +102,7 @@ def unpatch_method(cls, method_name):
     delattr(cls, DATADOG_NAMESPACE.format(method=method_name))
 
 def unpatch_cache():
-    cache_backends = [cache['BACKEND'] for cache in django_settings.CACHES.values()]
+    cache_backends = {cache['BACKEND'] for cache in django_settings.CACHES.values()}
```
[the remark this answers is not in the document]
Comment (wenbochang) [reply ddtrace/contrib/django/cache.py]: also changed

**Comment by brettlangdon** [comment]
(don't worry about the failed tests, something unrelated, already fixing)

**Comment by wenbochang** [comment]
@brettlangdon thanks so much for your fast responses!

**Comment by brettlangdon** [comment]
thanks again @wenbochang this will get released with the next version of `ddtrace` (sometime next week)

**Pull request merged**